tests/python/test_with_dask.py

```diff
@@ -0,0 +1,73 @@
+import testing as tm
+from distributed import Client, LocalCluster
+import dask.dataframe as dd
+import xgboost as xgb
+import numpy as np
+import pytest
+
+pytestmark = pytest.mark.skipif(**tm.no_sklearn())
+
+
+def run_train():
+    # Contains one label equal to rank
+    dmat = xgb.DMatrix([[0]], label=[xgb.rabit.get_rank()])
+    bst = xgb.train({"eta": 1.0}, dmat, 1)
+    return bst.predict(dmat)
+
+
+def test_train():
+    # Train two workers, the first has label 0, the second has label 1
+    # If they build the model together the output should be 0.5
+    cluster = LocalCluster(n_workers=2, threads_per_worker=2)
+    client = Client(cluster)
+    preds = xgb.dask.run(client, run_train)
+    assert all(v[0] == 0.5 for v in preds.values())
+
+
+def run_dask_dataframe(X, y, weights):
+    dmat = xgb.dask.create_worker_dmatrix(X, y, weight=weights)
+    # Expect this worker to get two partitions and concatenate them
+    assert dmat.num_row() == 50
+
+
+def test_dask_dataframe():
+    cluster = LocalCluster(n_workers=2, threads_per_worker=2)
+    client = Client(cluster)
+    n = 10
+    m = 100
+    partition_size = 25
+    X = dd.from_array(np.random.random((m, n)), partition_size)
+    y = dd.from_array(np.random.random(m), partition_size)

```

Comment:
You could also do `da.random.random(m, partition_size)` (though this isn't important).